docs/06-contributors/999-rfcs/000-stories/story-15.md

```diff
@@ -0,0 +1,186 @@
+    // is this the right synatx for multiple ops? 
+    this.inflights.add("set_status", ref: "this", op: ["_add", "get"]); 
+    this.inflights.add("set_estimation", ref: "this", op: ["_add", "get"]); 
+
+    // TODO add more permissions for remove, find
+  }
+
+  inflight get(id: str): Json {
+     return Json.parse(this._redis.GET(id));
+  }
+  
+  inflight _add(id: str, j: Json): str {
+    this._redis.SET(id , Json.format(j));
+    this._redis.SADD("todo", id);
+    return id;
+  } 
+  
+  inflight add(title: str): str {
+    // PLACEHOLDER - how does untyped works with numeric operations
+    let id = "${js.Math.floor(js.Math.random() * 100000000000)}";
+    let j = Json { 
+      title: title, 
+      status: Status.Uncompleted
+    };
+    print("adding task ${id} with data: ${j}"); 
+    return this_add(id, js);
+  }
+
+  inflight remove(id: str) {
+    print("removing task ${id}");
+    this._redis.DEL(id);
+  }
+
+  // PLACEHOLDER - having an untyped type
+  inflight find(term: str): Array<str> { 
+    let r = new js.RegExp(term);
+    let result = MutArray<str>[]; 
+    let ids = this._redis.SMEMBERS("todo");
+    for id in ids {
+      let j = Json.parse(this._redis.GET(id));
+      // Notice that there is autocasting from untyped to bool here 
+      if r.test(j.title) {
+        result.push(id);
+      }
+    }
+    return result.copy();
+  }
+
+  inflight set_status(id: str, status: Status): str {
+    let j = Json.clone_mut(this.get(id));
+    j.status = status;
+    this._add(id, Json.clone(j));
+    return id;
+  }
+
+  inflight set_estimation(id: str, estimation: duration): str {
+    let j = Json.clone_mut(this.get(id));
+    j.effort_estimation = estimation;
+    this._add(id, j);
+    return id;
+  }
+}
+
+resource TaskListApi {
+  api: cloud.Api;
+  task_list: ITaskList;
+  init(task_list: ITaskList) {
+    this.task_list = task_list;
+    this.api = new cloud.Api();
+    
+    // TODO add this.put
+    
+    this.api.post("/tasks", inflight (req: cloud. Api.ApiRequest): cloud.ApiResponse => {
+      let var title = str.from_json(req.body.title);
+      // Easter Egg - if you add a todo with the single word "random" as the title, 
+      //              the system will fetch a random task from the internet
+      if title == "random" {
+        // PLACEHOLDER - can I cast an untyped ?
+        let random_task = axios.get('https://www.boredapi.com/api/activity');
+        title = str.from_json(random_task.data.activity); 
+      } 
+      let id = this.task_list.add(title);
+      return cloud.ApiResponse { status:201, body: Json.format(id) };
+    });
+
+    this.api.get("/tasks/{id}", inflight (req: cloud.ApiRequest): cloud.ApiResponse => {
+      let id = str.from_json(req.params.id);
+      try {
+        let title = this.task_list.get(id);
+        return cloud.ApiResponse {status:200, body: Json.format(title)};
+      } catch {
+        return cloud.ApiResponse(status:400);
```

Comment:
Is the second syntax `cloud.ApiResponse` supported? I think only the first way of creating structs has been written right now - the potential issue I see with the second is that it makes `cloud.ApiResponse` look like a method call or a constructor (missing the `new` keyword).

Reply:
fixed, thanks